src/management/file_details.rs

```diff
@@ -0,0 +1,35 @@
+use std::fmt::Display;
+
+use anyhow::{bail, Result};
+use async_trait::async_trait;
+use reqwest::StatusCode;
+
+use crate::{client::FILES_ENDPOINT, upload::types::Response, ErrorResponse, ImageKit};
+
+#[async_trait]
+pub trait Details {
+    /// Given a file id retrieves the information within
+    async fn get_file_details<T: Display + Send>(&self, id: T) -> Result<Response>;
+}
+
+#[async_trait]
+impl Details for ImageKit {
+    async fn get_file_details<T: Display + Send>(&self, id: T) -> Result<Response> {
```

Comment:
Checking on https://docs.imagekit.io/api-reference/media-api/get-file-details I can see `fileId` is of type string, why not using `<S: AsRef<str>>`?. `Display` and `Send` are too generic for this purpose.

```suggestion
    async fn get_file_details<S: AsRef<str>>(&self, id: S) -> Result<Response> {
```

By using `<S: AsRef<str>>` we allow users to provide:

- `&str`
- `String`
- `&String`

Which is flexible yet safe enough!

Reply:
Seems to be right, thanks I'll get it fixed

Reply:
Would be cool! But please create a separate PR!